e2e/client/playwright/workspace.spec.ts

```diff
@@ -0,0 +1,156 @@
+import {test, expect} from '@playwright/test';
+import {Monitoring} from './page-object-models/monitoring';
+import {restoreDatabaseSnapshot, s} from './utils';
+
+test.describe('Custom Workspace', async () => {
+    test.describe('Create custom workspace', async () => {
+        test('save custom workspace', async ({page}) => {
+            await restoreDatabaseSnapshot();
+            await page.goto('/#/workspace/monitoring');
+
+            await page.locator(s('monitoring--selected-desk')).click();
+            await page.locator(s('create-workspace')).click();
+            await expect(page.locator(s('workspace-modal', 'workspace-name'))).toBeVisible();
+            await expect(page.locator('button', {hasText: 'save'})).toBeDisabled();
+            await page.locator(s('workspace-modal')).locator(s('workspace-name')).fill('new custom workspace');
+            await expect(page.locator('button', {hasText: 'save'})).toBeEnabled();
+            await page.locator('button', {hasText: 'save'}).click();
+            await expect(page.locator(s('monitoring--selected-desk'), {hasText: 'new custom workspace'})).toBeVisible();
+        });
+
+        test('cancel creating custom workspace', async ({page}) => {
+            await restoreDatabaseSnapshot();
+            await page.goto('/#/workspace/monitoring');
+
+            await page.locator(s('monitoring--selected-desk')).click();
+            await page.locator(s('create-workspace')).click();
+            await expect(page.locator(s('workspace-modal', 'workspace-name'))).toBeVisible();
+            await expect(page.locator('button', {hasText: 'save'})).toBeDisabled();
+            await page.locator(s('workspace-modal')).locator(s('workspace-name')).fill('new custom workspace');
+            await expect(page.locator('button', {hasText: 'cancel'})).toBeEnabled();
+            await page.locator('button', {hasText: 'cancel'}).click();
+            await expect(
+                page.locator(s('monitoring--selected-desk'), {hasText: 'new custom workspace'}),
+            ).not.toBeVisible();
+        });
+    });
+
+    test.describe('Create article in custom workspace', async () => {
+        test('Create article in custom workspace', async ({page}) => {
```

Comment:
You're testing too many things here. What is important in this test is whether you can create an article in a custom workspace.

What is not important for this test is:

- whether disabled state of a "save" button works
- whether it works to close an article

Those things are important in general, but they are a part of the authoring component and would be tested separatelly. For exampe you could also check if word count is displayed correctly, but it is out of scope for this test. Unless you think it's likely that it could work in desks, but not work in custom workspaces.